apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.tsx

```diff
@@ -401,93 +414,245 @@ onDocumentReady(() => {
-          <input
-            class="form-control time-limit-field"
-            type="number"
-            name="time_add"
-            aria-label="Time value"
-            style="width: 5em"
-            value="5"
-          />
-          <select class="custom-select time-limit-field" name="time_ref" aria-label="Time unit">
-            <option value="minutes">minutes</option>
-            ${row.time_remaining_sec !== null
-              ? html`<option value="percent">% total limit</option>`
-              : ''}
-          </select>
-        </p>
-        ${row.has_closed_instance
-          ? html`
-              <div class="form-check mb-2 reopen-closed-field">
+  function TimeLimitExplanation({ plusMinus }: { plusMinus: TimeLimitPlusMinus }) {
+    let explanation = '';
+    if (plusMinus === 'set_total') {
+      explanation =
+        'Updating the total time limit will set the given amount of time for the assessment based on when the assessment was started.';
+    } else if (plusMinus === 'set_rem') {
+      explanation =
+        'Updating the time remaining will set the given amount of time for the assessment based on the current time.';
+    } else if (plusMinus === 'set_exact') {
+      explanation = 'This will set the exact closing time for the assessment.';
+    } else if (plusMinus === '+1') {
+      explanation = 'This will add the given amount of time to the remaining time limit.';
+    } else if (plusMinus === '-1') {
+      explanation = 'This will subtract the given amount of time from the remaining time limit.';
+    } else if (plusMinus === 'unlimited') {
+      explanation = 'This will remove the time limit and the assessment will remain open.';
+    } else if (plusMinus === 'expire') {
+      explanation = 'This will expire the time limit and the assessment will be closed.';
+    }
+    return (
+      <small class="text-muted d-block" style="width:400px; max-width:90vw">
+        {explanation}
+      </small>
+    );
+  }
+
+  function TimeLimitEditPopover({
+    row,
+  }: {
+    row: {
+      action: string;
+      assessment_instance_id: number;
+      date: string;
+      has_closed_instance: boolean;
+      has_open_instance: boolean;
+      total_time: string;
+      time_remaining: string;
+      time_remaining_sec: number;
+      open: boolean;
+    };
+  }) {
+    // Because we need to update values on the Preact virtual DOM, we need to manage our
+    // form values as state. These can then be passed and used by the child components to
+    // display values or used for conditional rendering.
+    const [form, setForm] = useState<{
+      plus_minus: TimeLimitPlusMinus;
+      time_add: number;
+      date: string;
+      reopen_closed: boolean;
+      reopen_without_limit: boolean;
+    }>({
+      plus_minus: '+1',
+      time_add: 5,
+      date: Temporal.Now.zonedDateTimeISO(timezone).toPlainDateTime().toString().slice(0, 16),
+      reopen_closed: false,
+      reopen_without_limit: true,
+    });
```

Comment:
You can save yourself from having to explicitly specify the type for this object:

```suggestion
    const [form, setForm] = useState({
      plus_minus: '+1' as TimeLimitPlusMinus,
      time_add: 5,
      date: Temporal.Now.zonedDateTimeISO(timezone).toPlainDateTime().toString().slice(0, 16),
      reopen_closed: false,
      reopen_without_limit: true,
    });
```